### PR TITLE
geant4: v10.0.2 -> v10.4.1, define datasets seperately, add dependencies for enableInventor

### PR DESCRIPTION
**Pull request by tmplt**
###### Motivation for this change
Add another version of geant4 sources and define all related datasets in a seperate set of derivations.
Dependencies for geant4's `enableInventor` flag has been added (soxt).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Tmplt** [comment]
I've never seen the usage of `dependency ? null` before. I left them in because I was unsure: since most of them hint at libraries with the same name, shouldn't the `? null` just be omitted? After all, libraries can be overridden, right?

File: pkgs/development/libraries/soxt/default.nix
Comment (veprbl) [inline pkgs/development/libraries/soxt/default.nix]: Perhaps "A GUI binding for using Open Inventor with Xt/Motif"

Comment (Tmplt) [reply pkgs/development/libraries/soxt/default.nix]: That sounds good to me.

**Comment by veprbl** [comment]
This is coming together well. Could you please squash the geant4 changes after you are done with it?

File: pkgs/development/libraries/physics/geant4/datasets.nix
Comment (veprbl) [inline pkgs/development/libraries/physics/geant4/datasets.nix]: I have my doubts about this. I see that ```writeTextFile``` is not used for static files in ```pkgs/```, I wonder if it would be better to stick with regular file here as well.

File: pkgs/development/libraries/physics/geant4/default.nix
Comment (veprbl) [inline pkgs/development/libraries/physics/geant4/default.nix]: ditto

File: pkgs/development/libraries/physics/geant4/datasets.nix
Comment (Tmplt) [reply pkgs/development/libraries/physics/geant4/datasets.nix]: I figured since the setup hooks were to small they could just be inlined. Is `writeTextFile` meant to be used in another way?

**Comment by Tmplt** [comment]
I'll squash the geant4 changes when I figure out how to do it around the merges with master.

File: pkgs/development/libraries/physics/geant4/datasets.nix
Comment (veprbl) [reply pkgs/development/libraries/physics/geant4/datasets.nix]: I would not be surprised if there is some nasty corner case with it. I suppose I know one relatively harmless side-effect: if I'm not mistaken, the ```writeTextFile``` is a derivation and will "build" a new file for each architecture.

Comment (veprbl) [reply pkgs/development/libraries/physics/geant4/datasets.nix]: I kind of agree with you, but that discussion goes beyond the scope of this PR.

**Comment by Tmplt** [comment]
That should do it.

File: pkgs/development/libraries/physics/geant4/datasets.nix
Comment (veprbl) [inline pkgs/development/libraries/physics/geant4/datasets.nix]: don't need ```writeTextFile```

File: pkgs/development/libraries/physics/geant4/default.nix
Comment (veprbl) [inline pkgs/development/libraries/physics/geant4/default.nix]: ditto

File: pkgs/development/libraries/physics/geant4/default.nix
Comment (veprbl) [inline pkgs/development/libraries/physics/geant4/default.nix]: ditto

**Comment by Tmplt** [comment]
Thanks; fixed.

**Comment by veprbl** [comment]
@GrahamcOfBorg build geant4.v10_0_2 geant4.v10_4_1

**Review by veprbl — APPROVED** [review]
Checked against examples/basic/B1, was able to simulate events using data files. X11/OpenGl visual would not work for me, but thats probably due to my setup.

I will address some remaining issues with geant4 in a separate PR after this gets merged.

File: pkgs/development/libraries/soxt/default.nix
Comment (xeji) [inline pkgs/development/libraries/soxt/default.nix]: Would you like to add yourself as a maintainer here as well?

Comment (Tmplt) [reply pkgs/development/libraries/soxt/default.nix]: Might as well.

**Comment by xeji** [comment]
@GrahamcOfBorg build geant4.v10_0_2 geant4.v10_4_1 soxt

**Pull request merged**
